lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -60,6 +63,56 @@ class HighContrastSupport {
   }
 }
 
+/// Determines the [ui.AppLifecycleState] by listening to 'focus', 'blur', and
+/// 'beforeunload' events on the [DomWindow].
+class AppLifecycleStateSupport {
+  static AppLifecycleStateSupport instance = AppLifecycleStateSupport();
```

Comment:
info: Creation of this class cannot be done inline now that it needs a `_setAppLifecycleState` function tear-off